firedrake/preconditioners/pmg.py

```diff
@@ -529,8 +682,8 @@ def make_blas_kernels(self, Vf, Vc):
             return;
         }}
         """
-        self.prolong_kernel = op2.Kernel(prolong_code, "prolongation", ldargs=["-lpetsc"])
-        self.restrict_kernel = op2.Kernel(restrict_code, "restriction", ldargs=["-lpetsc"])
+        self.prolong_kernel = op2.Kernel(prolong_code, "prolongation", ldargs=["-lblas"])
```

Comment:
On supercomputers we usually get our BLAS from libraries other than `libblas.so`. Is there a way to figure out cleanly what BLAS we should be linking against here? I think the rationale for using PETSc was that it would be linked to the correct BLAS, but correct me if I'm wrong.

Reply:
@djanekovic might have some useful input on this one...

Reply:
This was bug I introduced and Pablo fixed. It is maybe confusing why we are including `petscblaslapack.h` but we need to link to `libblas.so`. That header file implements just the name mangling logic in order to call correct BLAS/LAPACK function. Because of that, we need to link to the `libblas.so` in the end.
Patrick, yes, you are correct and this problem was solved just recently. Jack added `get_petsc_variables` function in his PR #1928 which enables getting a path to the BLAS/LAPACK provided to the PETSc configure.
To sum up, Pablo, to link with correct BLAS library you should implement the same logic Jack implemented here: https://github.com/firedrakeproject/firedrake/blob/master/firedrake/slate/slac/compiler.py#L69

Reply:
You can import those variables from the slate file and then you need to do
```python
op2.Kernel(...,
                             include_dirs=BLASLAPACK_INCLUDE.split(),
                             ldargs=BLASLAPACK_LIB.split())
```

I think